portfolio/src/main/java/com/google/sps/servlets/CuisineDataServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.CuisineDatastore;
+import com.google.sps.data.AvailableCuisines;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+@WebServlet("/cuisine-data")
+public final class CuisineDataServlet extends HttpServlet {
+  private static final Logger log = Logger.getLogger(CuisineDataServlet.class.getName());
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Map<AvailableCuisines, Integer> cuisineVotes = CuisineDatastore.fetchCuisineVotes();
+   
+   // TODO: shouldn't be using .getLocalizedName() for the map key
+    Map<String, Integer> cuisineVotesLocalized = new HashMap<>();
+    for (AvailableCuisines cuisineId : cuisineVotes.keySet()) {
+      cuisineVotesLocalized.put(cuisineId.getLocalizedName(), cuisineVotes.get(cuisineId));
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(cuisineVotesLocalized);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /** Class used to parse the json object in post request body. */
+  public class UserCuisineVote {
+    private String userId;
+    private String cuisineId;
+
+    public String getUserId() {
+      return this.userId;
+    }
+
+    public String getCuisineId() {
+      return this.cuisineId;
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    UserCuisineVote cuisineVote = new Gson().fromJson(request.getReader(), UserCuisineVote.class);
```

Comment:
no super strong opinion but im not sure the UserCuisineVote class is necessary here. All you need is read userId, read cuisine Id from the request. the class structure is a bit overkill specially given that you unpack it immediately after reading it.